src/string.cr

```diff
@@ -891,11 +891,30 @@ class String
     end
   end
 
+  # Returns the `Char?` at the given *index*.
+  #
+  # Negative indices can be used to start counting from the end of the string.
+  #
+  # See '#[]'
+  #
+  # ```
+  # "hello"[0]  # => 'h'
+  # "hello"[1]  # => 'e'
+  # "hello"[-1] # => 'o'
+  # "hello"[-2] # => 'l'
+  # "hello"[5]  # => nil
+  # ```
   def []?(index : Int) : Char?
     char_at(index) { nil }
   end
 
-  def []?(str : String | Char)
+  # Returns the `String?` if the given *str* is in *self*.
```

Comment:
Which `String?`? Ditto for below.

Also the code fails to compile for `Char` arguments due to the added return value restriction. It shouldn't be added.

Reply:
updated via https://github.com/crystal-lang/crystal/pull/11690/commits/dfcf6bc6315cc5329d770c187425ca1c55499e11.  I dont know but I think this solves the concern you have.